Sources/MapboxNavigation/Route.swift

```diff
@@ -79,7 +79,15 @@ extension Route {
             })
         }
         
-        return hasRestriction ? features : []
+        if hasRestriction {
+            return features
```

Comment:
Because we map the coordinates of the features' geometry to generate the shape as the restricted route layer source. When it has restriction on the route, it will return multiple features as before. If it doesn't have restriction on the route, previously it will lead to a `LineString` with no coordinates inside, and cause the error message as `A line string must have two or more coordinate points.`. 
Right now, it will has a single feature with shape of the whole route, but the `RestrictedRoadClassAttribute` property as `false`. So the line gradient will apply the `UIColor.defaultTraversedRouteColor` to hide the whole restricted route layer to make it invisible, and also avoid the error message. And the `UIColor.defaultTraversedRouteColor` won't be customized to force hiding the whole restricted route layer, as in the unit test.